src/transformers/models/bert/modeling_bert.py

```diff
@@ -1517,14 +1518,29 @@ def forward(
 
         loss = None
         if labels is not None:
-            if self.num_labels == 1:
-                #  We are doing regression
+            if self.problem_type is None:
+                if self.num_labels == 1:
+                    self.problem_type = "regression"
+                elif self.num_labels > 1 and type(labels) == torch.long:
+                    self.problem_type = "single_label_classification"
+                else:
+                    self.problem_type = "multi_label_classification"
```

Comment:
The problem here is that it will be set as attribute to the model but not config attribute, so when saving the config, it won't be saved :-(

I think the easiest is to remove the shortcut `self.problem_type` and always use `self.config.problem_type`.

Reply:
done